modules/core/keeper/msg_server.go

```diff
@@ -751,7 +751,84 @@ func (k Keeper) ChannelUpgradeInit(goCtx context.Context, msg *channeltypes.MsgC
 
 // ChannelUpgradeTry defines a rpc handler method for MsgChannelUpgradeTry.
 func (k Keeper) ChannelUpgradeTry(goCtx context.Context, msg *channeltypes.MsgChannelUpgradeTry) (*channeltypes.MsgChannelUpgradeTryResponse, error) {
-	return nil, nil
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	module, _, err := k.ChannelKeeper.LookupModuleByChannel(ctx, msg.PortId, msg.ChannelId)
+	if err != nil {
+		ctx.Logger().Error("channel upgrade try failed", "port-id", msg.PortId, "error", errorsmod.Wrap(err, "could not retrieve module from port-id"))
+		return nil, errorsmod.Wrap(err, "could not retrieve module from port-id")
+	}
+	cbs, ok := k.Router.GetRoute(module)
+	if !ok {
+		ctx.Logger().Error("channel upgrade try failed", "port-id", msg.PortId, "error", errorsmod.Wrapf(porttypes.ErrInvalidRoute, "route not found to module: %s", module))
+		return nil, errorsmod.Wrapf(porttypes.ErrInvalidRoute, "route not found to module: %s", module)
+	}
+
+	channel, found := k.ChannelKeeper.GetChannel(ctx, msg.PortId, msg.ChannelId)
+	if !found {
+		return nil, errorsmod.Wrapf(channeltypes.ErrChannelNotFound, "port ID (%s) channel ID (%s)", msg.PortId, msg.ChannelId)
+	}
+
+	proposedUpgrade, err := k.ChannelKeeper.ChanUpgradeTry(
+		ctx,
+		msg.PortId,
+		msg.ChannelId,
+		msg.ProposedUpgradeFields,
+		msg.ProposedUpgradeTimeout,
+		msg.CounterpartyProposedUpgrade,
+		msg.CounterpartyUpgradeSequence,
+		msg.ProofChannel,
+		msg.ProofUpgrade,
+		msg.ProofHeight,
+	)
+	if err != nil {
+		ctx.Logger().Error("channel upgrade try failed", "error", errorsmod.Wrap(err, "channel handshake upgrade try failed"))
+		return &channeltypes.MsgChannelUpgradeTryResponse{
+			ChannelId:          msg.ChannelId,
+			Version:            proposedUpgrade.Fields.Version,
+			UpgradeSequence:    channel.UpgradeSequence,
```

Comment:
this value could be incorrect if the channel state was updated in `ChanUpgradeTry`. It looks like the response might be discarded in case of msg execution [failure](https://github.com/cosmos/cosmos-sdk/blob/main/baseapp/baseapp.go#L792)? Can we confirm if this response will be used?

Reply:
ahh, if we are writing an error receipt we should not be returning an error (state will be reverted)